samcli/local/lambdafn/runtime.py

```diff
@@ -353,9 +353,10 @@ def _clean_decompressed_paths(self):
         Clean the temporary decompressed code dirs
         """
         LOG.debug("Cleaning all decompressed code dirs")
-        for decompressed_dir in self._temp_uncompressed_paths_to_be_cleaned:
-            shutil.rmtree(decompressed_dir)
-        self._temp_uncompressed_paths_to_be_cleaned = []
+        with self._container_manager._lock:
```

Comment:
Could we make a different lock instead of using the one that the manager uses for image pulling? That way it doesn't mess with the image lock managing logic and the temp directory cleaning logic can happen separately.